editor/editor_node.cpp

```diff
@@ -2214,6 +2215,7 @@ void EditorNode::_edit_current(bool p_skip_foreign) {
 			NodeDock::get_singleton()->set_node(current_node);
 			SceneTreeDock::get_singleton()->set_selected(current_node);
 			InspectorDock::get_singleton()->update(current_node);
+			inspector_only = !inspector_only && node_select_toggles_inspector_only && editor_plugin_screen != nullptr && editor_plugin_screen->get_name() == "Script";
```

Comment:
```C++
inspector_only = !inspector_only
```
If `inspector_only` is true, this will always disable it.

```C++
editor_plugin_screen != nullptr && editor_plugin_screen->get_name() == "Script"
```
Instead of this, you could do `ScriptEditor::get_singleton()->is_visible_in_tree()` ([that's how it was done previously](https://github.com/godotengine/godot/blob/838cb598e3ef5572018d3d549c15bb27b0a5864d/editor/scene_tree_dock.cpp#L1384)).

Reply:
`inspector_only = !inspector_only`

My intention here is to not override `inspector_only` when it's already `true`.

Say `node_select_toggles_inspector_only == false` and `inspector_only == true`, then `inspector_only` will be overriden to `false` (this is based on the assumption that there are already situations where `inspector_only` may be true).

With my intention, we try to override it only when a context switching was already going to take place (`inspector_only == false`).

---

`ScriptEditor::get_singleton()->is_visible_in_tree()`

Cool, thanks.

Reply:
> stay_in_script_editor_on_node_selected 

@KoBeWi I also prefer your suggestion. Also, do you think `text_editor/behavior/navigation` is still the best category to put it?

Reply:
> My intention here is to not override inspector_only when it's already true.

Except with the current code, `inspector_only` will always be turned false if other conditions fail, because you used only `&&`.

I think the condition should be something like
```
inspector_only = inspector_only || (node_select_toggles_inspector_only && ...);
```

> do you think text_editor/behavior/navigation is still the best category to put it?

From the existing categories in `text_editor` group this one is the best.

Reply:
Sorry, I thought it was already there, I fixed exactly that (`||`) yesterday and haven't pushed it.

> From the existing categories in text_editor group this one is the best.
Cool, I'll push with the new name and the logic fix.